emitter/test/emitter-test-part-1.js

```diff
@@ -3,129 +3,101 @@ const EventEmitter = require('../emitter-part-1.js');
-        emitter.on("Hello", byOneTwo);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
-        
-        function byOneTwo() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
+        emitter.on("Hello", trackedFunction);
+        emitter.trigger("Hello");
+
+        tracker.verify();
+    });
+
+    it("supports several event handlers", function () {
+        const trackedFunction = tracker.calls(1);
+        const trackedFunction1 = tracker.calls(1);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("Hello", trackedFunction1);
         emitter.trigger("Hello");
+
+        tracker.verify();
     });
     
-    it("supports same function multiple times", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("Hello", byOne);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
-        
+    it("supports same function multiple times", function () {
+        const trackedFunction = tracker.calls(2);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("Hello", trackedFunction);
         emitter.trigger("Hello");
+
+        tracker.verify();
     });
 
-    it("Supports multiple event triggers same function", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("World", byOne);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
+    it("Supports multiple event triggers same function", function () {
+        const trackedFunction = tracker.calls(2);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction);
         emitter.trigger("Hello");
         emitter.trigger("World");
+
+        tracker.verify();
     });
 
-    it("Supports multiple event triggers different function", function (done) {
-        let counter = 0;
-        emitter.on("Hello", byOne);
-        emitter.on("World", byOneTwo);
-        function byOne() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
-
-        function byOneTwo() {
-            counter++;
-            if (counter === 2) {
-                done();
-            }
-        }
+    it("Supports multiple event triggers different function", function () {
+        const trackedFunction = tracker.calls(1);
+        const trackedFunction1 = tracker.calls(1);
+
+        emitter.on("Hello", trackedFunction);
+        emitter.on("World", trackedFunction1);
         emitter.trigger("Hello");
         emitter.trigger("World");
+
+        tracker.verify();
     });
 
-    it("Supports multiple events", function (done) {
+    it("Supports multiple events", function () {
```

Comment:
Let's turn all the functions to arrow functions? I think it's a bit distracting when it sometimes arrow and sometimes regular function